multipass-addressing/src/main/scala/com/precisely/bigdata/sample/spark/MultipassAddressing.scala

```diff
@@ -83,60 +80,83 @@ object MultipassAddressing {
     val verifyAddress = verifyResult.get(0).getAddress
-    
-    //  create multiLine request from the result fields obtained from verify api
-    val multiLineRequest: RequestAddress = new RequestAddress()
-    multiLineRequest.setStreet(verifyAddress.getStreet)
-    multiLineRequest.setAddressNumber(verifyAddress.getAddressNumber)
-    multiLineRequest.setAdmin1(verifyAddress.getAdmin1.getLongName)
-    multiLineRequest.setAdmin2(verifyAddress.getAdmin2.getLongName)
-    multiLineRequest.setCountry(verifyAddress.getCountry.getIsoAlpha3Code)
-    multiLineRequest.setCity(verifyAddress.getCity.getLongName)
-    multiLineRequest.setPostalCode(verifyAddress.getPostalCode)
-
-    val multiLineResponse = addressing.geocode(multiLineRequest, preferences.orNull)
+
+    var multiLineRequest: RequestAddress = input.requestAddress()
+
+    var singleLine = input.requestAddress().getStreet + " " +
+      input.requestAddress().getCity + " " +
+      input.requestAddress().getAdmin1 + " " +
+      input.requestAddress().getPostalCode
+
+    if (CollectionUtils.isNotEmpty(verifyResult)) {
+      //  create multiLine request from verify Response
+      multiLineRequest = new RequestAddress()
+      multiLineRequest.setStreet(verifyAddress.getStreet)
+      multiLineRequest.setAddressNumber(verifyAddress.getAddressNumber)
+      multiLineRequest.setAdmin1(verifyAddress.getAdmin1.getLongName)
+      multiLineRequest.setAdmin2(verifyAddress.getAdmin2.getLongName)
+      multiLineRequest.setCountry(verifyAddress.getCountry.getIsoAlpha3Code)
+      multiLineRequest.setCity(verifyAddress.getCity.getLongName)
+      multiLineRequest.setPostalCode(verifyAddress.getPostalCode)
+
+      // create singleline request from verify Response
+      singleLine = verifyAddress.getFormattedAddress + " " + verifyAddress.getCountry.getIsoAlpha3Code
+    }
+
+    val multiLineResponse = addressing.geocode(multiLineRequest, relaxedGeocoderPreferences)
     val multiLineResult = multiLineResponse.getResults
 
-    if (CollectionUtils.isNotEmpty(multiLineResult) && multiLineResult.get(0).getCustomFields
-      .get("PRECISION_CODE") != null && Seq("S5", "S6", "S7", "S8").contains(multiLineResult
-      .get(0).getCustomFields.get("PRECISION_CODE").slice(0, 2))) {
-      return multiLineResponse
+    if (CollectionUtils.isNotEmpty(multiLineResult) && multiLineResult.get(0).getLocation.getExplanation.getType != null) {
+      if ("ADDRESS_POINT".equals(multiLineResult.get(0).getLocation.getExplanation.getType.label)) {
+        return multiLineResponse
+      }
     }
 
-    //  create singleline request from multiLine Response
-    val singleLine = verifyAddress.getFormattedAddress + " " + verifyAddress.getCountry.getIsoAlpha3Code
-    val singleLineRequest: RequestAddress = new RequestAddress();
+    val singleLineRequest: RequestAddress = new RequestAddress()
     singleLineRequest.setAddressLines(Arrays.asList(singleLine))
```

Comment:
For a single item, I'd use Collections.singletonList() over Arrays.asList()

Reply:
Yes, I also think that will be better choice. Thanks